project1/server/server_tasks/agent_task.cpp

```diff
@@ -19,10 +19,10 @@ namespace server_tasks {
         if (read_manager_ && write_manager_) {
             // give the agent a unique file handler with the shared access managers
             auto fh = std::make_unique<server::file_handler::ThreadSafeFileHandler>(read_manager_, write_manager_);
-            agent_ = new server::Agent(client_fd_, std::move(fh), active_commands_);
+            agent_ = std::make_unique<server::Agent>(client_fd_,std::move(fh),std::move(active_commands_));
```

Comment:
I'm honestly not sure what `std::move`ing a shared pointer does, but it doesn't make a lot of sense in this case. I think you should just pass it normally. (For `fh` it is valid.)

Reply:
There is actually an error when I pass only `fh`...